### ISSUE TITLE
fzn-or-tools: UNSATISFIABLE for satisfiable models

### ISSUE THREAD

**Issue by ceandrade** [google/or-tools#1957]
Dear all,

fzn-or-tools is returning UNSATISFIABLE for model/data that are actually satisfiable (a mixed integer programming model). 

First, we create a flatzinc file using the OR Tools predicates:

```bash
$ ~/minizinc-2.1.7/bin/mzn2fzn -G or-tools model.mzn data.dzn --fzn model.fzn --ozn model.ozn
```

Then, using  or-tools_CentOS-7.3.1611-64bit_v6.4.4495, we can use the flag "-use_cp_sat=false", and we have the solution we want:

```bash
$ ~/or-tools_CentOS-7.3.1611-64bit_v6.4.4495/bin/fzn-or-tools -statistics -use_cp_sat=false -num_solutions=1 -threads=12 -time_limit=120000 model.fzn
%% Start search (memory used = 226.42 MB)
%% Root node processed (time = 27 ms, constraints = 339, memory used = 226.42 MB)
%% Solution #0 (objective value = 19907366390, time = 39 ms, branches = 136, failures = 0, depth = 100, memory used = 226.43 MB, limit = 0%)
loaderAssignment = array3d(1..12, 1..3, 1..5, [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 320, 0, 0, 320, 0, 320, 0, 0, 0, 0, 0, 320, 320, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 320, 320, 0, 0, 0, 0, 0, 0, 1, 0, 320, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 99, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 320, 0, 0, 100, 0, 0, 320, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 320, 0, 0, 0, 320, 0, 320, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0, 320, 0, 0, 0, 0, 0]);
----------
%% Finished search tree (time = 119996 ms, branches = 6410145, failures = 3255369, memory used = 226.43 MB)
...
...
```

However, if we drop option "use_cp_sat", we cannot solve the problem: 

```bash
$ ~/or-tools_CentOS-7.3.1611-64bit_v6.4.4495/bin/fzn-or-tools -statistics -num_solutions=1 -threads=12 -time_limit=120000 model.fzn
=====UNSATISFIABLE=====
```

The most recent version is even worse since it looks like we cannot use option "use_cp_sat":

```bash
$ ~/or-tools_flatzinc_v7.5.7466/bin/fzn-or-tools -statistics -use_cp_sat=false -num_solutions=1 -threads=12 -time_limit=120000 model.fzn
ERROR: unknown command line flag 'use_cp_sat'
```

I have attached the FZN file for appreciation (with the txt extension; please change to fzn):

[model.txt](https://github.com/google/or-tools/files/4457338/model.txt)

**Comment by lperron** [comment]
Thanks for the bug report.
We dropped support for the original CP solver support as performance was way worse.

We will have a look.

Thanks

**Comment by ceandrade** [comment]
Great, thanks @lperron. I would like to have som cycles to help you guys with that, but I cannot at this moment. Please, keep me posted. Cheers.

**Comment by lperron** [comment]
OK, the issue is that we trim integer variables to int32min..int32max. With the fix, the problem is solved is 0.05 seconds with 8 threads.

This is pushed on master.
